ReactiveFeedback/Reducer.swift

```diff
@@ -0,0 +1,27 @@
+public typealias Reducer<State, Event> = (State, Event) -> State
+
+public func combine<State, Event>(
+    _ reducers: Reducer<State, Event>...
+) -> Reducer<State, Event> {
+    return { state, event in
+        return reducers.reduce(state) { (newState, reducer) -> State in
+            return reducer(newState, event)
+        }
+    }
+}
+
+public func pullback<LocalState, GlobalState, LocalEvent, GlobalEvent>(
+    _ reducer: @escaping Reducer<LocalState, LocalEvent>,
+    value: WritableKeyPath<GlobalState, LocalState>,
+    event: WritableKeyPath<GlobalEvent, LocalEvent?>
+) -> Reducer<GlobalState, GlobalEvent> {
```

Comment:
While I know this impl comes from @pointfreeco , IMO:

1. It's better to rename `value` parameter name to `state` for more readability.
2. [nit] I'm not sure if "pullback" is really a good term here. This term is also used as contravariant-functor-mapping in SnapshotTesting, and I don't know if this reducer transformation could also become a part of some more generalised "pullback" concept. (I would personally prefer renaming to `transform` to avoid confusion)

P.S. 
2's "pullback" comes from categorical terminology which @mbrandonw definitely has some intuitions around this.
It will be great if you can share some thoughts on this naming (and possible relation with category theory) @mbrandonw !

Reply:
This is a perfectly fine use of the term `pullback`, it just isn't happening in the category of types and functions like we are used to 😄. Let's forget about the name `pullback` for a moment and just call it `contramap`.

If **Type** is the category of types and functions, then "contramap" is what Haskell defines to express the idea of a contravariant functor:

```
F: Type -> Type

contramap: (A -> B) -> F<B> -> F<A>
```

But there are lots of categories out there, not just the boring category of types and functions. We could also consider **Type_kp**, the category of types where a morphism from `A` to `B` is a key path `WritableKeyPath<A, B>`. In this category, `contramap` would have the form:

```
F: Type_kp -> Type

contramap: WritableKeyPath<A, B> -> F<B> -> F<A>
```

Note that `F` is not an endofunctor, which what we usually see in programming world. Instead `F` is going from one category to a different one, and that is why `contramap` takes a `WritableKeyPath` yet returns a plain function.

But it still has the same basic shape as the regular `contramap`. It transforms some process that goes from `A` to `B` into another process that goes from `B` to `A`.

And it still satisfies all the properties a contravariant functor needs to satisfy. In particular:

```
contramap(\.self) == id 

contramap(\.foo.bar) == contramap(\.foo) <<< contramap(\.bar)
```

So mathematically, this is a perfectly fine thing to do. We are just defining contravariant functors on different categories. We of course can only use this category knowledge in order to guide us, there is no way to actually get a handle on this category from the perspective of the Swift type system. Even Haskell struggles with grappling with these other "exotic" categories. 

So, with all of that said, whether you want to use `pullback`, `contramap`, `transform` or something else (back in the day I called it `lift` 🤭) is up to you. I particularly like the intuition `pullback` gives, and think it works well regardless if you are pulling back reducers, predicates, comparators, snapshot strategies, or whatever!

Reply:
@mbrandonw
Thanks for clear explanation 👏👏👏 
The idea of thinking `WritableKeyPath<A, B>` (and possibly many of optics types) as an arrow in different category was very mind-blowing to me, although I knew it is composable but I couldn't identify it as same as arrow unfortunately 😅

So, I'm now +1 for having `pullback` as the consistent name now!